src/radical/pilot/agent/launch_method/jsrun.py

```diff
@@ -121,20 +121,26 @@ def _create_resource_set_file(self, slots, uid, sandbox):
         # "error in ptssup_mkcltsock_afunix()"
         rs_str  = 'cpu_index_using: logical\n'
 
-        rank_id = 0
-        for rank in slots['ranks']:
-
-            gpu_maps = list(rank['gpu_map'])
-            for map_set in rank['core_map']:
-                cores = ','.join(str(core) for core in map_set)
-                rs_str += 'rank: %d: {' % rank_id
-                rs_str += ' host: %s;'  % str(rank['node_id'])
-                rs_str += ' cpu: {%s}'  % cores
-                if gpu_maps:
-                    gpus = [str(gpu_map[0]) for gpu_map in gpu_maps]
-                    rs_str += '; gpu: {%s}' % ','.join(gpus)
-                rs_str  += ' }\n'
-                rank_id += 1
+        base_id = 0
+        for ranks_per_slot in slots['ranks']:
```

Comment:
same comment as above

Reply:
JSRun definitely should know and understand the mapping, since it maps to "resource set" concept. Ok to rename `ranks_per_slot` if it brings some confusion